org.mwc.debrief.lite/src/main/java/org/mwc/debrief/lite/gui/custom/JSelectTrackModel.java

```diff
@@ -140,14 +141,29 @@ public void setPrimaryTrack(final TrackWrapper newPrimary)
   }
 
   @Override
-  public void setTracks(final List<TrackWrapper> tracks)
+  public boolean setTracks(final List<TrackWrapper> tracks)
   {
+    boolean isDifferent = false;
     final List<TrackWrapperSelect> oldTracks = this._tracks;
-    this._tracks = new ArrayList<>();
+    final List<TrackWrapperSelect> newTracks = new ArrayList<>();
+    final HashSet<TrackWrapper> oldTracksSet = new HashSet<>();
+    if (oldTracks != null)
+    {
+      for (TrackWrapperSelect oldTrack : oldTracks)
+      {
+        oldTracksSet.add(oldTrack.track);
+      }
+    }
     for (final TrackWrapper track : tracks)
     {
-      this._tracks.add(new TrackWrapperSelect(track, false));
+      newTracks.add(new TrackWrapperSelect(track, false));
+      isDifferent |= !oldTracksSet.contains(track);
+    }
+    if (isDifferent)
+    {
+      this._tracks = newTracks;
```

Comment:
Hmm, with a slight change, we can make _tracks a final field.

So, instead of:
````
this._tracks = newTracks;
````
we would use:
````
 _tracks.clear();
 _tracks.addAll(newTracks);
````

There isn't a strong argument for the change, it's just the slight benefit of the compiler ensuring _tracks never gets deleted/replaced.

@saulhidalgoaular - what do you think?

Reply:
I agree. I am changing it too.